modules/ecs-service/main.tf

```diff
@@ -0,0 +1,696 @@
+    Version = "2012-10-17"
+    Statement = [{
+      Effect = "Allow"
+      Principal = {
+        Service = "ecs-tasks.amazonaws.com"
+      }
+      Action = "sts:AssumeRole"
+      Condition = {
+        test     = "ArnLike"
+        variable = "aws:SourceArn"
+        values = [
+          "arn:aws:ecs:${data.aws_region.current.name}:${data.aws_caller_identity.current.account_id}:*"
+        ]
+      }
+      Condition = {
+        test     = "StringEquals"
+        variable = "aws:SourceAccount"
+        values = [
+          data.aws_caller_identity.current.account_id
+        ]
+      }
+    }]
+  })
+}
+
+resource "aws_iam_role_policy_attachment" "task" {
+  count = length(var.ecs_task_role_policy_arns) > 0 ? length(var.ecs_task_role_policy_arns) : 0
+
+  role       = aws_iam_role.task[0].name
+  policy_arn = var.ecs_task_role_policy_arns[count.index]
+}
+
+
+# -------------------------------------------
+# CREATE THE TASK EXECUTION IAM ROLE
+# -------------------------------------------
+
+resource "aws_iam_role" "task_execution" {
+  name_prefix = "${var.ecs_service_name}-execution"
+
+  assume_role_policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [{
+      Effect = "Allow"
+      Principal = {
+        Service = "ecs-tasks.amazonaws.com"
+      }
+      Action = "sts:AssumeRole"
+    }]
+  })
+}
+
+
+# -------------------------------------------
+# CREATE THE SECRETS MANAGER TASK EXECUTION IAM POLICY
+# -------------------------------------------
+
+locals {
+  # A list of resource ARNs that will be authorized in the
+  # iam policy for the task execution role.
+  secrets_manager_arns = compact(
+    concat(
+      [
+        for k, v in var.ecs_container_secrets :
+        # Removes the key references from the SecretsManager ARNs
+        replace(v, ":${k}::", "")
+      ],
+      [var.docker_credential_secretsmanager_arn]
+    )
+  )
+}
+
+resource "aws_iam_policy" "secrets_manager" {
+  count = length(var.ecs_container_secrets) > 0 || var.docker_credential_secretsmanager_arn != "" ? 1 : 0
+
+  name = "${var.ecs_service_name}-secretsmanager-read"
+  policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [{
+      Effect = "Allow"
+      Action = [
+        "secretsmanager:GetSecretValue"
+      ]
+      Resource = local.secrets_manager_arns
+    }]
+  })
+}
+
+
+# -------------------------------------------
+# ATTACH POLICIES TO THE TASK EXECUTION IAM ROLE
+# -------------------------------------------
+
+resource "aws_iam_role_policy_attachment" "secrets_manager" {
+  count = length(var.ecs_container_secrets) > 0 || var.docker_credential_secretsmanager_arn != "" ? 1 : 0
+
+  role       = aws_iam_role.task_execution.name
+  policy_arn = aws_iam_policy.secrets_manager[0].arn
+}
+
+resource "aws_iam_role_policy_attachment" "task_execution" {
+  role       = aws_iam_role.task_execution.name
+  policy_arn = "arn:aws:iam::aws:policy/service-role/AmazonECSTaskExecutionRolePolicy"
```

Comment:
This is a default AWS Roles?

Reply:
That is a default AWS policy that the task execution needs.